### ISSUE TITLE
Adding ssh passphrase to obsidian md

### ISSUE THREAD

**Issue by Tom-there** [denolehov/obsidian-git#245]
First of all not really sure if this is the right place to ask this, sorry if it isn't

So I just set up Obsidian git for my main Windows machine and my arch laptop, and on my windows machine it works perfectly,
but since I have a ssh passphrase on my arch key the plugin gets denied permission, how do I fix this?
(I read through the other issues already but didn't really understand the steps taken. My ssh agent is set up correctly, should I just generate a key without passphrase solely for obsidian?)

**Comment by tintin10q** [comment]
I think he means this message:

![image](https://user-images.githubusercontent.com/24190849/183078187-fee9720d-790c-4194-8b7c-9bdb4ce7206d.png)

I get this because I can't fill in a ssh key phrase. Is there any way to solve this?

**Comment by acmorse** [comment]
> I think he means this message:
> 
> ![image](https://user-images.githubusercontent.com/24190849/183078187-fee9720d-790c-4194-8b7c-9bdb4ce7206d.png)
> 
> I get this because I can't fill in a ssh key phrase. Is there any way to solve this?

I think I had the same problem, and I got around it by creating an SSH key without a passphrase. It's probably not ideal, but at least it works for now...

**Comment by tintin10q** [comment]
Yes that is the message I meant. I guess that makes sense. I don't really want to make a key without a pass phrase though but it probably would be hard to detect? Anyway I think the best way to do it is to have a hotkey which when pressed asks for the passfhrase and then it is helt in memory until you restart.

**Comment by Tom-there** [comment]
sorry for not checking the thread for so long...
This is the exact error I meant, for now I created a ssh-key without a passphrase, so its working now

**Comment by tintin10q** [comment]
Could you reopen this? Because it would still be nice if we were able to put a passphrase.

**Comment by Vinzent03** [comment]
I'm trying to summarize authorization instructions [here](https://github.com/denolehov/obsidian-git/wiki/Authentication). It may help you.

**Comment by kg4zow** [comment]
Does the plugin not work with `ssh-agent`?

**Comment by tintin10q** [comment]
What is that?

**Comment by kg4zow** [comment]
"What is that?" referring to `ssh-agent`?

`ssh-agent` is a process which holds secret keys in memory and uses them to process authentication requests. Any operation which needs a secret key and would normally prompt you for a passphrase, can be performed by `ssh-agent` instead.

For most people the attraction is that you only get asked for the key's passphrase once, when the key is added to the agent (which usually happens the first time you use a key after the `ssh-agent` process starts).

There are other programs out there which speak the same protocol, and can perform the same operations using different mechanisms. For example, `gpg-agent` can emulate an SSH agent and use a GPG subkey to authenticate SSH connections. Even better, `gpg-agent` can work by talking to an external smartcard (or in my case, a YubiKey) which actually contains the secret keys and performs the crypto operations with it, without the keys ever leaving the card.

I *could* go on about this, but I don't have the time. If this is *really* new to you, you may need to spend some time with a search engine, or with the documentation for your SSH client.

**Comment by jsco2t** [comment]
I'm also having the same issue. I *have* looked at the auth instructions (https://github.com/denolehov/obsidian-git/wiki/Authentication). I'm on MacOS. My terminal can access the repo in question without issue using my ssh key (key password is stored in the keychain). The `gitconfig` for the repo in question has:

```
[credential]
	helper = osxkeychain
```

Yet I still get:

<img width="389" alt="image" src="https://github.com/denolehov/obsidian-git/assets/7284382/3f8ec61e-9073-4d83-9e41-1ca6c67313fe">

Any suggestions as to what I might be doing wrong here?

**Comment by jsco2t** [comment]
To be clear - I don't necessarily need a way to store my passphrase in Obsidian. I just need to work out why this plugin on MacOS seems to have issues accessing my credentials in the OSX Keychain.
